src/helpers/optionalStacking/common.ts

```diff
@@ -28,24 +37,111 @@ export function getViewByTwoAttributes(
+    const measuresItems = findBucketByLocalIdentifier(clonedBuckets, MEASURES).items || [];
+
+    if (measuresItems.length > 1) {
+        return {
+            config,
+            buckets: disableComputeRatio(clonedBuckets),
+        };
+    }
+
+    if (measuresItems.length === 1) {
+        const isComputeRatio = getComputeRatio(measuresItems[0]);
+        const hasStackByAttribute = isStackedChart(clonedBuckets);
+        const newBuckets =
+            isComputeRatio && hasStackByAttribute ? ignoreStackBucket(clonedBuckets) : clonedBuckets;
+        const { stackMeasures, stackMeasuresToPercent } = config;
+
+        return {
+            buckets: newBuckets,
+            config: {
+                ...config,
+                stackMeasures: stackMeasures && !isComputeRatio,
+                stackMeasuresToPercent: stackMeasuresToPercent && !isComputeRatio && hasStackByAttribute,
+            },
+        };
+    }
+
+    return {
+        buckets,
+        config,
+    };
+}
+
+function disableComputeRatio(buckets: VisualizationObject.IBucket[]): VisualizationObject.IBucket[] {
+    return buckets.map((bucket: VisualizationObject.IBucket) => {
+        if (bucket.localIdentifier === MEASURES) {
+            const items: VisualizationObject.BucketItem[] = (bucket.items || []).map(
+                (item: VisualizationObject.BucketItem) => {
+                    if (getComputeRatio(item)) {
+                        setComputeRatio(item, false);
+                    }
+                    return item;
+                },
+            );
+            return {
+                ...bucket,
+                items,
+            };
+        }
+
+        return bucket;
+    });
+}
+
+function ignoreStackBucket(buckets: VisualizationObject.IBucket[]): VisualizationObject.IBucket[] {
+    return buckets.map((bucket: VisualizationObject.IBucket) => {
+        if (bucket.localIdentifier === STACK) {
+            return {
+                ...bucket,
+                items: [],
+            };
+        }
+
+        return bucket;
+    });
+}
+
+export function getComputeRatio(bucketItem: VisualizationObject.BucketItem): boolean {
+    return get(bucketItem, ["measure", "definition", "measureDefinition", "computeRatio"], false);
+}
+
+function setComputeRatio(bucketItem: VisualizationObject.BucketItem, value: boolean) {
+    set(bucketItem, ["measure", "definition", "measureDefinition", "computeRatio"], value);
+}
+
+/**
+ * Show a measure as a percentage
+ * one measure
+ *      => computeRatio = true (ignore stackBy)
+ *      => computeRatio = false or not set => evaluate stackMeasuresToPercent
+ * multiple measures
+ *      => ignore computeRatio and use stackMeasuresToPercent
+ *
+ * stackMeasures is applied only when there are [2 measures and more]
+ * stackMeasuresToPercent is applied only when there are [1 measure + 1 stackBy] or [2 measures and up]
+ */
+export function getSanitizedStackingConfigForVisualization(
```

Comment:
name of function should reflect its functionality NOT the place where it is used. I would rename it to something like 'getSanitizedStackingConfigFromAfm'

Reply:
updated